src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -135,6 +138,23 @@ class CodyToolWindowContent(private val project: Project) {
     allContentLayout.show(allContentPanel, MAIN_PANEL)
   }
 
+  private fun selectHistory(state: ChatState) {
+    val session =
+        AgentChatSession.getSessionByInternalId(state.internalId!!)
+            ?: AgentChatSession.createFromState(project, state)
+    addChatSession(session)
+  }
+
+  private fun deleteHistory(state: ChatState) {
+    HistoryService.getInstance().remove(state.internalId!!)
+    val session = AgentChatSession.getSessionByInternalId(state.internalId!!)
+    if (session != null) {
+      AgentChatSession.removeSession(session)
+      val isVisible = session.getPanel().chatSession.getInternalId() == state.internalId!!
```

Comment:
`session.getPanel().chatSession.getInternalId()` is kind of internal info of the UI panel, we are doing it a bit backwards I think. Maybe we should just add `AtomicReference<ChatSession> currentSession` here in `CodyToolWindowContent` and keep it up to date?
In the long term reaching deep into objects like that will make the code spaghetti again :P

Reply:
Fixed by Piotr.